### PR TITLE
[web] Get the size from visualviewport instead of window.innerHeight/innerW…

### PR DESCRIPTION
**Pull request by nturgut**
get the size from visualviewport instead of window.innerHeight/innerWidth.Change safari resize. If visualviewport API is supported these values are the same. visualviewport dimensions are reliable on safari so don't setup a resize timer.

Tested in:
- Safari on IOS (13)
- Safari on MacOs (13)
- Chrome on Linux Desktop
- Chrome on Android
- Firefox  on Linux Desktop

Also confirmed that for the current implementation on android on chrome, the innerheight (used in calculating the physical size) value changes with the visual keyboard.

Fixes: https://github.com/flutter/flutter/issues/42211 for Safari 13

### CODE REVIEW

**Comment by nturgut** [comment]
> LGTM if you've tested it in the flutter gallery and everything works. Few cases to try:
> 
> * Resizing the window.
> * Zoom-in and zoom-out.
> * Opening and closing devtools.

Thanks @mdebbar for the suggestions. Initially I tested 1,3 and also opening closing keyboard on mobile. After your comment I also tested the zoom in/out

All seems the same. I also debug the values. If viewport API is enabled window.innerHeight = visualviewportHeight, same true for width.

File: lib/web_ui/lib/src/engine/dom_renderer.dart [hunk 1]
```diff
@@ -393,26 +393,31 @@ flt-glass-pane * {
...
-          // Window size changed. Notify.
-          t.cancel();
-          _metricsDidChange(null);
-        } else if (checkCount > 5) {
-          // Checked enough times. Stop.
-          t.cancel();
-        }
-      });
+    if (html.window.visualViewport == null) {
+      if (browserEngine == BrowserEngine.webkit) {
```
[the remark this answers is not in the document]
Comment (nturgut) [reply lib/web_ui/lib/src/engine/dom_renderer.dart]: Done. Thanks!

File: lib/web_ui/lib/src/engine/dom_renderer.dart [hunk 2]
```diff
@@ -393,12 +393,16 @@ flt-glass-pane * {
     // is 1.0.
     window.debugOverrideDevicePixelRatio(1.0);
 
-    if (browserEngine == BrowserEngine.webkit) {
+    if (html.window.visualViewport == null &&
```
Comment (yjbanov) [inline lib/web_ui/lib/src/engine/dom_renderer.dart]: Let's leave a comment explaining the possibilities that trigger the `if` block, including Firefox.

Comment (nturgut) [reply lib/web_ui/lib/src/engine/dom_renderer.dart]: done.

**Comment by nturgut** [comment]
Submitting this PR. It's was green until the comment change. Hopefully, will start another LUCI build, making the tree green.

**Pull request merged**